src/influx_udp.erl

```diff
@@ -8,71 +6,119 @@
+-export([start/0, stop/0]).
 
--export([start_link/0]).
+-export([start_link/0, init_server/0]).
 
 -export([
+  start_pool/2,
   write/1,
-  write/2
+  write/2,
+  write/3
 ]).
 
 %% ------------------------------------------------------------------
 %% gen_server Function Exports
 %% ------------------------------------------------------------------
 
--export([init/1, handle_call/3, handle_cast/2, handle_info/2,
+-export([handle_call/3, handle_cast/2, handle_info/2,
          terminate/2, code_change/3]).
 
+-record(state,
+{
+  defaults ::map() %% default pool configuration
+}).
+
 %% ------------------------------------------------------------------
 %% API Function Definitions
 %% ------------------------------------------------------------------
 
 start_link() ->
-  gen_server:start_link({local, ?SERVER}, ?MODULE, [], []).
-
+  proc_lib:start_link(?SERVER, init_server, []).
 
 %% ===================================================================
 %% Application callbacks
 %% ===================================================================
 
 start() ->
-  start([],[]).
+  application:start(influx_udp).
 
-start(_StartType, _StartArgs) ->
-  influx_udp_sup:start_link().
-
-stop(_State) ->
-    ok.
+stop() ->
+  application:stop(influx_udp).
 
 %% @doc
-%% Write binary data to influx.
-%% Note: Assumed that data represents Influx-valid JSON.
+%% Start new pool with Name and Options (as map)
+%% Options:
+%% - host - InfluxDB hostname
+%% - port - InflxuDB UDP port
+%% - pool_size - Pool size (numer of workers)
+%% - max_overflow - Pool max overflow size
 %% @end
+-spec start_pool(Name::atom(), Options::map()) -> ok.
+start_pool(Name, Options) ->
+  gen_server:call(?SERVER, {start_pool, Name, Options}).
 
+%% @doc
+%% Write binary data to influx using default pool.
+%% Note: Assumed that data represents InfluxDB-valid input.
+%% @end
 -spec write(Bin::binary()) -> ok.
-
 write(Bin) ->
-  gen_server:call(?SERVER, {send_to_pool, {write, Bin}}).
+  gen_server:call(?SERVER, {send_to_pool, default, {write, Bin}}).
 
 %% @doc
-%% Write data points to Series.
+%% Write binary data to influx using named pool.
 %% @end
+-spec write(Pool::atom(), Bin::binary()) -> ok
+    ; (
+        Series::atom()|string()|binary(),
+        Points::list(influx_data_point())|influx_data_point()
+      ) -> ok.
+write(Pool, Bin) when is_binary(Bin) ->
+  gen_server:call(?SERVER, {send_to_pool, Pool, {write, Bin}});
 
--spec write(Series::atom()|string()|binary(), Points::list(influx_data_point())|influx_data_point()) -> ok.
+%% @doc
+%% Write data points to Series using default pool.
+%% @end
 write(Series, Points) ->
-  gen_server:call(?SERVER, {send_to_pool, {write, Series, Points}}).
+  gen_server:call(?SERVER, {send_to_pool, default, {write, Series, Points}}).
+
+%% @doc
+%% Write data points to Series using named pool.
+%% @end
+-spec write(Pool::atom(), Series::atom()|string()|binary(), Points::list(influx_data_point())|influx_data_point()) -> ok.
```

Comment:
Line 89 is too long: -spec write(Pool::atom(), Series::atom()|string()|binary(), Points::list(influx_data_point())|influx_data_point()) -> ok..